crates/router/src/core/cache.rs

```diff
@@ -0,0 +1,20 @@
+use common_utils::errors::CustomResult;
+
+use super::errors;
+use crate::{
+    cache::CONFIG_CACHE,
+    db::StorageInterface,
+    services::{self},
+};
+
+pub async fn invalidate(
+    store: &dyn StorageInterface,
+    key: &str,
+) -> CustomResult<services::api::ApplicationResponse<serde_json::Value>, errors::ApiErrorResponse> {
+    CONFIG_CACHE.remove(key).await;
+
+    match store.get_redis_conn().delete_key(key).await {
+        Ok(_) => Ok(services::api::ApplicationResponse::StatusOk),
+        _ => Err(errors::ApiErrorResponse::RemoveRedisKeyFailure.into()),
```

Comment:
Preferably, we should be returning a response with 404 status code in case a key was not found, and a response with 500 status code if a Redis related issue occurred.

Reply:
I am happy to return the InternalServerError(HTTP 500) as part of the error. 

In case of the key does not exist what needs to be done may be up for discussion. As of now RedisConnectionPool's `del(key)` encapsulation states  

> Removes the specified keys. A key is ignored if it does not exist. --> Based on this currently we are returning HTTP 200 status code. 

If we need to return the HTTP 404 status code,  need to extend `RedisConnectionPool::delete_key`.  This has currently been referenced in a couple of places.

Reply:
@dracarys18 @Narayanbhat166 What should be the behavior here?

Reply:
We can change this function https://github.com/juspay/hyperswitch/blob/cef8914372fa051f074e89fc76b76c6aee0d7bca/crates/redis_interface/src/commands.rs#L151 
to return a https://github.com/juspay/hyperswitch/blob/cef8914372fa051f074e89fc76b76c6aee0d7bca/crates/redis_interface/src/errors.rs#L52 error, and then the caller can match based on the error type?

Reply:
@SanchithHegde @jeevaprakashdr As I can see in the fred documentation del will ignore if key does not exist in the database so it won't throw an error, So if del failed possibility is that something went wrong with redis so makes sense to throw InternalServerError.

Ref: https://docs.rs/fred/latest/fred/interfaces/trait.KeysInterface.html#method.del

Reply:
@dracarys18 That seems to be how Redis handles it as well (official docs for the [`DEL` command](https://redis.io/commands/del/)). However, since we are only deleting one key, the output of the `DEL` command as an integer must be either 0 or 1. If we absolutely need to return a 404, we can match over the response (0 or 1), and return an appropriate result (`Ok(())` or `Err(NotFound)`). If we get any other response than 0 or 1, we'd have to error out stating an unexpected response from Redis.

Reply:
> @dracarys18 That seems to be how Redis handles it as well (official docs for the [`DEL` command](https://redis.io/commands/del/)). However, since we are only deleting one key, the output of the `DEL` command as an integer must be either 0 or 1. If we absolutely need to return a 404, we can match over the response (0 or 1), and return an appropriate result (`Ok(())` or `Err(NotFound)`). If we get any other response than 0 or 1, we'd have to error out stating an unexpected response from Redis.

This seems to be a good option.

Reply:
> @SanchithHegde @jeevaprakashdr As I can see in the fred documentation del will ignore if key does not exist in the database so it won't throw an error, So if del failed possibility is that something went wrong with redis so makes sense to throw InternalServerError.
> 
> Ref: https://docs.rs/fred/latest/fred/interfaces/trait.KeysInterface.html#method.del

Returning 200 will be ambiguous in this case, you are not sure whether the key was deleted or the key was given in the wrong format ( key not found ). It would be better if the end user is known that key was not found and hence the deletion did not happen